libraries/lib-music-information-retrieval/MusicInformationRetrieval.h

```diff
@@ -10,12 +10,38 @@
 **********************************************************************/
 #pragma once
 
+#include "MirTypes.h"
+
+#include <functional>
 #include <optional>
 #include <string>
+#include <unordered_map>
 #include <vector>
 
 namespace MIR
 {
+class MirAudioReader;
+struct BeatInfo;
+
+struct LoopClassifierSettings
+{
+   /*!
+    * False positive rate allowed for the classifier.
+    */
+   const double allowedFalsePositiveRate;
+
+   /*!
+    * Classifier score threshold above which the analyzed audio file can be
+    */
+   const double threshold;
```

Comment:
If I correctly understand the statistical methods, then whether the allowed false positive rate is consistent with the threshold, depends on probability distributions of ... whatever `threshold` measures, in two different populations of sound files, those that have rhythmic content and those that do not.

So then are the four magic numbers below derived experimentally from a certain corpus of example files that you take as representative?

Reply:
Two distributions of the score returned by the algorithm, yes, one for files whose ground truths say they are not music loops, and the other for the others. If the distributions do not overlap, then you have a perfect classifier and the threshold can be set anywhere between the two. But ours do, and we set the threshold such that it splits the non-loop population in 99% / 1% for the strict case, and 90% / 10% for the lenient case, the upper part of the split being the false-positive area.
Representativeness is indeed a topic. Ideally I could take a representative sample set from what users are importing, but I don't have this. So I need a data set I'm happy with, and believe in it. The one use in this PR (next commit) is worth what it's worth. A follow-up ticket, #5795, may provide something more representative (although the goal of that ticket is first and foremost to make the dataset available to others for reproducibility).